ikarus/python/dirichletvalues/dirichletvalues.hh

```diff
@@ -57,53 +115,22 @@ void registerDirichletValues(pybind11::handle scope, pybind11::class_<DirichletV
-                .first;
+  auto lv_      = Dune::Python::insertClass<LocalViewWrapper>(
+                 scopedf, "LocalView",
+                 Dune::Python::GenerateTypeName("Dune::Python::LocalViewWrapper", Dune::MetaType<Basis>()), includes)
+                 .first;
 
   cls.def(pybind11::init([](const Basis& basis) { return new DirichletValues(basis); }), pybind11::keep_alive<1, 2>());
 
-  // Eigen::Ref needed due to https://pybind11.readthedocs.io/en/stable/advanced/cast/eigen.html#pass-by-reference
-  cls.def("fixBoundaryDOFs",
-          [](DirichletValues& self, const std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int)>& f) {
-            auto lambda = [&](BackendType& vec, const MultiIndex& indexGlobal) {
-              // we explicitly only allow flat indices
-              f(vec.vector(), indexGlobal[0]);
-            };
-            self.fixBoundaryDOFs(lambda);
-          });
-
-  cls.def("fixBoundaryDOFsUsingLocalView",
-          [](DirichletValues& self,
-             const std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&)>& f) {
-            auto lambda = [&](BackendType& vec, int localIndex, LocalView& lv) {
-              auto lvWrapper = LocalViewWrapper(lv.globalBasis());
-              // this can be simplified when
-              // https://gitlab.dune-project.org/staging/dune-functions/-/merge_requests/418 becomes available
-              pybind11::object obj = pybind11::cast(lv.element());
-              lvWrapper.bind(obj);
-              f(vec.vector(), localIndex, lvWrapper);
-            };
-            self.fixBoundaryDOFs(lambda);
-          });
-
-  cls.def(
-      "fixBoundaryDOFsUsingLocalViewAndIntersection",
-      [](DirichletValues& self,
-         const std::function<void(Eigen::Ref<Eigen::VectorX<bool>>, int, LocalViewWrapper&, const Intersection&)>& f) {
-        auto lambda = [&](BackendType& vec, int localIndex, LocalView& lv, const Intersection& intersection) {
-          auto lvWrapper = LocalViewWrapper(lv.globalBasis());
-          // this can be simplified when
-          // https://gitlab.dune-project.org/staging/dune-functions/-/merge_requests/418 becomes available
-          pybind11::object obj = pybind11::cast(lv.element());
-          lvWrapper.bind(obj);
-          f(vec.vector(), localIndex, lvWrapper, intersection);
-        };
-        self.fixBoundaryDOFs(lambda);
-      });
+  cls.def_property_readonly("container", &DirichletValues::container, pybind11::return_value_policy::reference);
+  cls.def_property_readonly("size", &DirichletValues::size);
+  cls.def_property_readonly("fixedDOFsize", &DirichletValues::fixedDOFsize);
+  cls.def("isConstrained", [](DirichletValues& self, std::size_t i) -> bool { return self.isConstrained(i); });
+  cls.def("setSingleDOF", [](DirichletValues& self, std::size_t i, bool flag) { self.setSingleDOF(i, flag); });
```

Comment:
This should the als has `Basis::MultiIndex` as argument. In Python this can then by called as `setSingleDof((i))` or you keep this function change the cpp callsite looks like `self.setSingleDOF({i}, flag)`  and add a new function that takes s `Basis::MultiIndex`  with the old cpp callsite   `self.setSingleDOF(i, flag)`  .